File: R/statistics_with_R/03_R_Enviroment/Script_Files/04_readingData.R
# reading from a dat file
lecturerData <- read.delim('/home/atrides/Desktop/Applied-Statistics-with-R-master/statistics_with_R/03_learningR/Lecturer Data.dat', header=TRUE)
lecturerData

?gl
job<- gl(2,5,length=2*5,labels=c('lecturers','students'))
job


# refactoring

job <- factor(job, levels=levels(job)[c(2,1)])
job

lecturerData

# saving Data
write.csv(lecturerData, 'df.csv')

# manipulating data
newData <- lecturerData[, c('friends','income','neurotic')]
newData

# lecturer only
lecturer <- lecturerData[job=='lecturers',]
lecturer

#  personality  variables  but  only  for  people  who  drink  more  than  10  units  of alcohol
alcoholPersonality <- lecturerData[lecturerData$alcohol>10, c("friends", "alcohol", "neurotic")]
alcoholPersonality

# Dataframe to matrix
alcoholPersonalityMatrix <- as.matrix(alcoholPersonality)
alcoholPersonalityMatrix

File: Python/statistics_with_Python/03_Python_Enviroment/Script_Files/04_readingData.py
import pandas as pd
import numpy as np

lecturerData = pd.read_csv('/home/atrides/Desktop/R/statistics_with_Python/03_Python_Enviroment/Data_Files/Lecturer Data.dat', sep='\s+')


print(lecturerData)


lecturerData['job'] = lecturerData['job'].replace({1:'lecturer', 2:'students'})


# saving data
lecturerData.to_csv('final.csv')


# manipulating Data
print(lecturerData[['friends', 'income', 'neurotic']])

# lecturer only
print(lecturerData[lecturerData['job']=='lecturer'])


#  personality  variables  but  only  for  people  who  drink  more  than  10  units  of alcohol
print(lecturerData[lecturerData['alcohol']>10][["friends", "alcohol", "neurotic"]])



# Dataframe to matrix/array
print(np.array(lecturerData[lecturerData['alcohol']>10][["friends", "alcohol", "neurotic"]]))



